src/Tools/dotnet-monitor/CollectionRules/CollectionRuleContainer.cs

```diff
@@ -54,11 +54,64 @@ public async ValueTask DisposeAsync()
         {
             if (DisposableHelper.CanDispose(ref _disposalState))
             {
-                _disposalTokenSource.SafeCancel();
+                await StopRulesCore(CancellationToken.None);
+            }
+        }
+
+        public async Task StartRulesAsync(
+            IReadOnlyCollection<string> ruleNames,
+            CancellationToken token)
+        {
+            DisposableHelper.ThrowIfDisposed<CollectionRuleContainer>(ref _disposalState);
```

Comment:
Consider an assert or failure if we ever call Start twice in a row instead of Start/Stop.